romancal/source_catalog/source_catalog.py

```diff
@@ -1073,6 +1085,118 @@ def _split_skycoord(self, table):
+            and descriptions for PSF photometry catalog.
+        """
+
+        return [
+            {
+                "old_name": "flags",
+                "new_name": "flag_psf",
+                "desc": "Data quality flags",
+            },
+            {
+                "old_name": "x_fit",
+                "new_name": "x_psf",
+                "desc": "X coordinate as determined by PSF fitting",
+            },
+            {
+                "old_name": "x_err",
+                "new_name": "x_psf_err",
+                "desc": "Error on X coordinate of PSF fitting",
+            },
+            {
+                "old_name": "y_fit",
+                "new_name": "y_psf",
+                "desc": "Y coordinate as determined by PSF fitting",
+            },
+            {
+                "old_name": "y_err",
+                "new_name": "y_psf_err",
+                "desc": "Error on Y coordinate of PSF fitting",
+            },
+            {
+                "old_name": "flux_fit",
+                "new_name": "flux_psf",
+                "desc": "Source flux as determined by PSF photometry",
+            },
+            {
+                "old_name": "flux_err",
+                "new_name": "flux_psf_err",
+                "desc": "Source flux error as determined by PSF photometry",
+            },
+        ]
+
+    @lazyproperty
+    def psf_photometry_colnames(self) -> List:
+        """
+        Update and return column descriptions for PSF photometry results.
+
+        This method updates the column descriptions with PSF photometry-related information
+        and returns a list of column names.
+
+        Returns
+        -------
+            List of column names for PSF photometry results.
+        """
+        desc = {x["new_name"]: x["desc"] for x in self.psf_photometry_catalog_mapping}
+
+        self.column_desc.update(desc)
+
+        return list(desc.keys())
+
+    def do_psf_photometry(self) -> None:
+        """
+        Perform PSF photometry by fitting PSF models to detected sources for refined astrometry.
+
+        This method constructs a gridded PSF model based on instrument and detector information.
+        It then fits the PSF model to the image model's sources to improve astrometric precision.
+
+        """
+        log.info("Constructing a gridded PSF model.")
+        if hasattr(self.model.meta, "instrument"):
+            # ImageModel (L2 datamodel)
+            filt = self.model.meta.instrument.optical_element
+            detector = self.model.meta.instrument.detector.replace("WFI", "SCA")
+        else:
+            # MosaicModel (L3 datamodel)
+            filt = self.model.meta.basic.optical_element
+            detector = "SCA02"
+        gridded_psf_model, _ = psf.create_gridded_psf_model(
+            path_prefix="tmp",
+            filt=filt,
+            detector=detector,
+            overwrite=True,
+            logging_level="ERROR",
+        )
```

Comment:
Dave reported an issue where the temporary gridded model was not cleaned up.  Can you fix this if it's still an issue, or create an issue to fix it in the future?

Reply:
The issue has been fixed.